File: src/main/java/io/github/malczuuu/iemu/domain/firmware/FirmwareUpdateExecution.java
package io.github.malczuuu.iemu.domain.firmware;

import io.github.malczuuu.iemu.infrastructure.lwm2m.FirmwareUpdateResult;
import io.github.malczuuu.iemu.infrastructure.lwm2m.FirmwareUpdateState;

public interface FirmwareUpdateExecution {

  FirmwareUpdateExecution execute();

  String getPackageVersion();

  boolean hasNext();

  byte[] getFile();

  String getPackageUri();

  FirmwareUpdateState getState();

  FirmwareUpdateResult getResult();

  int getProgress();
}

File: src/main/kotlin/io/github/malczuuu/iemu/domain/firmware/FirmwareUpdateExecution.kt
package io.github.malczuuu.iemu.domain.firmware

import io.github.malczuuu.iemu.infra.lwm2m.FirmwareUpdateResult
import io.github.malczuuu.iemu.infra.lwm2m.FirmwareUpdateState

interface FirmwareUpdateExecution {

  val file: ByteArray

  val packageUri: String?

  val packageVersion: String?

  val state: FirmwareUpdateState

  val result: FirmwareUpdateResult

  val progress: Int

  fun execute(): FirmwareUpdateExecution

  fun hasNext(): Boolean
}
